### PR TITLE
Store encrypted votes, decrypt on election close

### PR DESCRIPTION
**Pull request by KonradSwierczynski**

### CODE REVIEW

File: backend/apollo/elections/models/election.py [hunk 1]
```diff
@@ -44,6 +47,17 @@ def can_be_opened(self) -> bool:
...
 
+    def _open_votes(self):
+        for vote in Vote.objects.filter(question__election=self):
+            try:
+                decrypted_answer_id = decrypt(self.secret_key, vote.answer_ciphertext)
+                answer = Answer.objects.get(pk=decrypted_answer_id)
+                if answer.question.election == self:
+                    vote.answer = answer
+                    vote.save()
+            except (CryptoError, Answer.DoesNotExist):
```
Comment (adambudziak) [inline backend/apollo/elections/models/election.py:58]: Are we going to have any mechanism to verify (with zero knowledge) that a vote is indeed for one of the election's questions?

Also, I'd move the `if` to `try except else`.

File: backend/apollo/elections/serializers.py [hunk 2]
```diff
@@ -53,46 +53,23 @@ class VoteSerializer(
...
 
     class Meta:
         model = Vote
-        fields = ["id", "answer", "author"]
+        fields = ["id", "answer_ciphertext", "author", "question"]
         read_only_fields = ["id"]
         validators = [
             UniqueTogetherValidator(
-                queryset=Vote.objects.all(), fields=["author", "answer"]
+                queryset=Vote.objects.all(), fields=["author", "question"]
```
Comment (adambudziak) [inline backend/apollo/elections/serializers.py:69]: I'd remove it - we're going to support multi-choice answers.

File: backend/apollo/elections/models/election.py (hunk 1 above)
Comment (KonradSwierczynski) [reply backend/apollo/elections/models/election.py]: We have to discuss how to do it first.

Good idea, I moved it

File: backend/apollo/elections/serializers.py (hunk 2 above)
Comment (KonradSwierczynski) [reply backend/apollo/elections/serializers.py:69]: We can do it after vote overwriting or after adding multiple-choice questions to the frontend

**Pull request merged**